File: 121_Best_Time_To_Buy_And_Sell_Stock/Solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        l = 0
        r = 1
        maxP = 0
        while r < len(prices):
            if prices[l] > prices[r]:
                l = r
            else:
                curProf = prices[r]-prices[l]
                maxP = max(curProf, maxP)
                r+=1

        
        return maxP

File: 121_Best_Time_To_Buy_And_Sell_Stock/Solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int left = 0;
        int right = 1;
        int maxProfit = 0;
        while(right < prices.size()){
            if(prices[left]>prices[right]){
                left = right;
            } else {
                int currentProfit = prices[right]-prices[left];
                maxProfit = max(currentProfit,maxProfit);
                right+=1;
            }
        }
        return maxProfit;
    }
};

